pkg/api/admin/openshiftcluster.go

```diff
@@ -48,7 +48,7 @@ type OpenShiftClusterProperties struct {
 	IngressProfiles                 []IngressProfile        `json:"ingressProfiles,omitempty"`
 	Install                         *Install                `json:"install,omitempty"`
 	StorageSuffix                   string                  `json:"storageSuffix,omitempty"`
-	RegistryProfiles                []RegistryProfile       `json:"registryProfiles,omitempty"`
+	RegistryProfiles                []RegistryProfile       `json:"registryProfiles,omitempty" mutable:"true"` // Not actually mutable! We discard it in ToInternal, this just means that any changes do not cause static validation to fail.
```

Comment:
I do not feel comfortable making this change for several reasons:

1. It blows my mind when someting is marked as mutable, but not actually mutable.  First intuitive response - this is not right.
2. The fact that we have to make this change indicates that there is an issue somewhere else. Where is it? Let's find the root casuse. Probably we just need to adjust the test in `pkg/frontend/openshiftcluster_putorpatch_test.go` after introducing a static validator there.
3. Making it mutable here means that it is now possible to modify registry profiles and pass the static validation which sounds dangerous. Most likely we are going to be fine with the existing code (with converter not converting the thing into internal representation), but this makes easier to introduce a regression.

---

Side note: we should not rely on testing mutability in `pkg/frontend/openshiftcluster_putorpatch_test.go`: it is extra (more like integration test). Mutability should be tested in the validator. And it looks like we are missing a test case for `registryProfiles` there.

Could you please test case into `TestOpenShiftClusterStaticValidateDelta` in `pkg/api/admin/openshiftcluster_validatestatic_test.go` while you are looking into it?

    ```go
		    {
			    name: "registryProfiles change is not allowed",
			    oc: func() *OpenShiftCluster {
				    return &OpenShiftCluster{
					    Properties: OpenShiftClusterProperties{
						    RegistryProfiles: []RegistryProfile{{Name: "test", Username: "testuser"}},
					    },
				    }
			    },
			    modify: func(oc *OpenShiftCluster) {
				    oc.Properties.RegistryProfiles[0].Username = "someothertestuser"
			    },
			    wantErr: "400: PropertyChangeNotAllowed: properties.registryProfiles: Changing property 'properties.registryProfiles' is not allowed.",
		    },
    ```

Reply:
> Probably we just need to adjust the test in `pkg/frontend/openshiftcluster_putorpatch_test.go` after introducing a static validator there.

I think the issue is that "patch a cluster with registry profile should ignore registry profile" test case does not reflect the actual behaviour: `wantStatusCode: http.StatusOK`. In fact the real API will return 400 Bad Request becuase of the static validator and this is what we want.